src/Ogooreck.Sample.BusinessLogic.FSharp.Tests/Deciders/BankAccountDecider.fs

```diff
@@ -0,0 +1,74 @@
+﻿module Deciders.BankAccountDecider
+
+open System
+open BankAccount
+open Deciders.BankAccount
+
+type OpenBankAccount =
+    { BankAccountId: Guid
+      AccountNumber: string
+      ClientId: Guid
+      CurrencyISOCode: string }
+
+type RecordDeposit = { Amount: decimal; CashierId: Guid }
+type WithdrawnCashFromATM = { Amount: decimal; AtmId: Guid }
+type CloseBankAccount = { Reason: string }
+
+type Command =
+    | OpenBankAccount of OpenBankAccount
+    | RecordDeposit of RecordDeposit
+    | WithdrawnCashFromATM of WithdrawnCashFromATM
+    | CloseBankAccount of CloseBankAccount
+
+let openBankAccount now command =
+    { BankAccountId = command.BankAccountId
+      AccountNumber = command.AccountNumber
+      ClientId = command.ClientId
+      CurrencyISOCode = command.CurrencyISOCode
+      CreatedAt = now ()
+      Version = 1 }
+
+let recordDeposit now (command: RecordDeposit, bankAccount) =
+    if (bankAccount.Status = BankAccountStatus.Closed) then
+        invalidOp "Account is closed!"
+
+    { BankAccountId = bankAccount.Id
+      Amount = command.Amount
+      CashierId = command.CashierId
+      RecordedAt = now ()
+      Version = bankAccount.Version + 1L }
+
+let withdrawCashFromATM now (command, bankAccount) =
+    if (bankAccount.Status = BankAccountStatus.Closed) then
+        invalidOp "Account is closed!"
+
+    if (bankAccount.Balance < command.Amount) then
+        invalidOp "Account is closed!"
+
+    { BankAccountId = bankAccount.Id
+      Amount = command.Amount
+      ATMId = command.AtmId
+      RecordedAt = now ()
+      Version = bankAccount.Version + 1L }
+
+let closeBankAccount now (command, bankAccount) =
+    if (bankAccount.Status = BankAccountStatus.Closed) then
+        invalidOp "Account is already closed!"
+
+    { BankAccountId = bankAccount.Id
+      Reason = command.Reason
+      ClosedAt = now ()
+      Version = bankAccount.Version + 1L }
+
+let decide now command bankAccount =
+    match command with
+    | OpenBankAccount c -> openBankAccount now c |> BankAccountOpened
+    | RecordDeposit c ->
+        recordDeposit now (c, bankAccount)
+        |> DepositRecorded
```

Comment:
I would put the the DepositRecorded case construction directly in the function, so it would return an event.

Reply:
Yeah, makes sense, good call. Done it in https://github.com/oskardudycz/Ogooreck/pull/13/commits/aea6cc181bb9d935f498d6f0499e0970f061da25.